app/assets/javascripts/app/views/publisher/uploader_view.js

```diff
@@ -57,23 +57,23 @@ app.views.PublisherUploader = Backbone.View.extend({
     var progress = Math.round(loaded / total * 100);
     this.info.text(fileName + " " + progress + "%").fadeTo(200, 1);
     this.publisher.photozoneEl
-      .find("li.loading").first().find(".progress-bar")
+      .find("li.loading#upload-" + id + " .progress-bar")
       .width(progress + "%");
   },
 
-  submitHandler: function() {
+  submitHandler: function(id) {
     this.$el.addClass("loading");
-    this._addPhotoPlaceholder();
+    this._addPhotoPlaceholder(id);
   },
 
   // add photo placeholders to the publisher to indicate an upload in progress
-  _addPhotoPlaceholder: function() {
+  _addPhotoPlaceholder: function(id) {
     var publisher = this.publisher;
     publisher.setButtonsEnabled(false);
 
     publisher.wrapperEl.addClass("with_attachments");
     publisher.photozoneEl.append(
-      "<li class=\"publisher_photo loading\" style=\"position:relative;\">" +
+      "<li id=\"upload-" + id + "\" class=\"publisher_photo loading\" style=\"position:relative;\">" +
```

Comment:
This id comes from fine-uploader and is a number, but please confirm it can't come from the user and there is no injection here as I don't know how fine-uploader works internally.

Reply:
The `id` is the index in the array, so it's safe to use it:

https://github.com/FineUploader/fine-uploader/blob/5.13.0/client/js/upload-data.js#L73-L79